scapy/tools/automotive/isotpscanner.py

```diff
@@ -0,0 +1,156 @@
+#! /usr/bin/env python
+
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Nils Weiss <nils@we155.de>
+# Copyright (C) Alexander Schroeder <alexander1.schroeder@st.othr.de>
+# This program is published under a GPLv2 license
+
+import argparse
+from argparse import RawTextHelpFormatter
+
+import scapy.modules.six as six
+from scapy.config import conf
+from scapy.consts import LINUX
+
+
+if six.PY2 or not LINUX:
+    conf.contribs['CANSocket'] = {'use-python-can': True}
+
+from scapy.contrib.cansocket import CANSocket, PYTHON_CAN  # noqa: E402
+from scapy.contrib.isotp import ISOTPScan  # noqa: E402
+
+
+def do_scan(socket, start_id, end_id, extended, noise_listen_time,
+            sniff_time, piso, interface_string, verbose):
+    print("Start scan (%s - %s)" % (hex(start_id), hex(end_id)))
+    result = ISOTPScan(socket,
+                       range(start_id, end_id + 1),
+                       extended_addressing=extended,
+                       noise_listen_time=noise_listen_time,
+                       sniff_time=float(sniff_time) / 1000,
+                       output_format="code" if piso else "text",
+                       can_interface=interface_string,
+                       verbose=verbose)
+    if piso:
+        result = "import can\n" \
+                 "conf.contribs['CANSocket'] = {'use-python-can': %s}\n" \
+                 "load_contrib('cansocket')\n" \
+                 "load_contrib('isotp')\n" % PYTHON_CAN + result
+    return result
+
+
+def main():
+    extended = False
+    extended_only = False
+    piso = False
+    verbose = False
+    parser = argparse.ArgumentParser(
```

Comment:
Could you stick to getopt (even if the calls are heavier), so that it works with Py2 natively ?

Reply:
`argparse` is a little bit more comfortable in my opinion. Is it really necessary to replace it with getopt? 
If you prefer not having another dependency in scapy, I can replace it.

Reply:
did it